src/util/Permissions.js

```diff
@@ -32,14 +32,28 @@ class Permissions {
 
   /**
    * Gets all given permissions that are missing from the bitfield.
-   * @param {PermissionResolvable[]} permissions Permissions to check for
+   * @param {PermissionResolvable[]|Permissions} permissions Permissions to check for
    * @param {boolean} [checkAdmin=true] Whether to allow the administrator permission to override
    * @returns {PermissionResolvable[]}
    */
   missing(permissions, checkAdmin = true) {
+    if (permissions instanceof Permissions) return this.missing(permissions.toArray());
     return permissions.filter(p => !this.has(p, checkAdmin));
   }
 
+  /**
+   * Returns this object as an {@link Array} of {@link PermissionResolvable}.
+   * @param {boolean} [checkAdmin=true] Whether to allow the administrator permission to override
+   * @returns {PermissionResolvable[]}
+   */
+  toArray(checkAdmin = true) {
+    let arr = [];
+    for (let perm in Permissions.FLAGS) {
```

Comment:
`for (const perm of Object.keys(Permissions.FLAGS) {` would be better here.

Reply:
Correction, ignore this comment.

Reply:
Nonetheless it's a good practise to check if the property exists, check the [`guard-for-in`](https://eslint.org/docs/rules/guard-for-in) rule from ESLint.

Reply:
Nonetheless it's a good practise to check if the property exists, check the [`guard-for-in`](https://eslint.org/docs/rules/guard-for-in) rule from ESLint.

Reply:
I don't believe the serialize method guards the for in loop, and it doesn't seem necessary for three simple object that is Permissions.FLAGS